reactive/zookeeper.py

```diff
@@ -0,0 +1,60 @@
+'''
+Reactive handlers for Zookeeper.
+'''
+
+from charmhelpers.core import hookenv
+from charms.layer.zookeeper import Zookeeper
+from charms.reactive import set_state, when, when_not
+
+@when('bigtop.available')
+@when_not('zookeeper.installed')
+def install_zookeeper():
+    '''
+    After Bigtop has done the initial setup, trigger a puppet install,
+    via our Zooekeeper library.
+
+    puppet will start the service, as a side effect.
+
+    '''
+    zk = Zookeeper()
+    zk.install()
+    zk.open_ports()
+    set_state('zookeeper.installed')
+    set_state('zookeeper.started')
+    hookenv.status_set('active', 'Ready')
+
+@when('zookeeper.started', 'config.changed.rest')
+def rest_config():
+    """TODO"""
+    pass
+
+@when('zookeeper.started', 'zkpeer.joined')
+def quorum_add(zkpeer):
+    """Add a zookeeper peer.
+
+    Add the unit that just joined, restart Zookeeper, and remove the
+    '.joined' state so we don't fall in here again (until another peer joins).
+    """
+    nodes = zkpeer.get_nodes()  # single node since we dismiss .joined below
+    zk = Zookeeper()
+    zk.increase_quorum(nodes)
+    #restart_zookeeper_if_config_changed()  # Not necessary due to puppet
+    zkpeer.dismiss_joined()
```

Comment:
This *almost* works. I get what looks like the right stuff written out the the zoo.cfg, and the server restarts.

However, `/usr/lib/zookeeper/bin/zkServer.sh status` doesn't work. It claims to have an error contacting the service, but the service is running.

I bet that it's something subtlely wrong with the way that I'm writing the config file. Haven't figured out what it is yet, though :-/